lib/Sema/ConstraintSystem.cpp

```diff
@@ -1071,12 +1071,12 @@ Type ConstraintSystem::replaceInferableTypesWithTypeVars(
         return openUnboundGenericType(unbound->getDecl(), unbound->getParent(),
                                       locator, /*isTypeResolution=*/false);
       } else if (auto *placeholderTy = type->getAs<PlaceholderType>()) {
-        if (auto *placeholderRepr = placeholderTy->getOriginator()
-                                        .dyn_cast<PlaceholderTypeRepr *>()) {
```

Comment:
This change needs an additional check for `PlaceholderTypeRepr` as is we only want to open `_` type reprs and not the invalid reprs.